### DISCUSSION TITLE
cannot see the gui

### DISCUSSION THREAD

**Discussion by xevilstar** [baresip/baresip#1953] [Q&A]
baresip
baresip v1.0.0 Copyright (C) 2010 - 2020 Alfred E. Heggestad et al.
Local network address:  IPv4=eth0|192.168.0.171 
aucodec: PCMU/8000/1
aucodec: PCMA/8000/1
ausrc: alsa
auplay: alsa
medianat: stun
medianat: turn
medianat: ice
Populated 0 accounts
account: No SIP accounts found
 -- check your config or add an account using 'uanew' command
Populated 3 contacts
Populated 2 audio codecs
Populated 0 audio filters
Populated 0 video codecs
Populated 0 video filters
baresip is ready.

**Comment by sreimers** [comment]
If you need a gui you have to enable the related application module (like `gtk.so`):
https://github.com/baresip/baresip/wiki/Configuration#loading-modules

Btw. version 1.0.0 is quite old. The latest release is v2.5.0

**Reply by jasonblewis** [reply]
I have enabled the module but it appears that gtk.so is not gettinb built for me by the standard make command.

is there anything special i need to do to enable this?

**Reply by jasonblewis** [reply]
nm, worked it out.

```
sudo apt-get install build-essential libgtk-3-dev
cmake .
make
sudo make install
ldconfig
```

**Comment by xevilstar** [comment]
done thanks

**Comment by ivanwmco** [comment]
I have the same problem. I installed and got baresip+ working on Android. Now I want to get baresip-gtk installed and working on Ubuntu. The baresip-gtk package on Ubuntu 22.04 is still version 1.0.0 and even with module_app gtk.so enabled in config it doesn't open a GUI. Just a green phone icon in the right top menu (with network, power, etc.) which lists account, status, dial, dial contact, about and quit.

Where can I find clear instructions on how to build the latest version (currently 3.13.0) of baresip and GTK into a baresip-gtk package for Ubuntu 22.04? I am new to all this, but willing to give it a serious try...

**Reply by mbattista** [reply]
Hi,

the green phone in the top right is the GUI.
If you press dial or get called a new Window opens.
The green phone will get a red phone if you have a missed call. 

The GUI only offers basic phone features and should be used in combination with the terminal menu and not as a replacement.

**Reply by ivanwmco** [reply]
Thanks for your swift response, mbattista. I will just have to get familiar with baresip's cli. That in itself shouldn't be too hard.

I would still like some clear instructions on how to build the latest version of baresip-gtk (version 3.13.0 instead of 1.0.0 as in the Ubuntu 22.04 repositories) or be redirected to where I can find such instructions. 

EDIT:
I added a linphone sip address in config:

> <sip:xxxxxxxxxx@sip.linphone.org;transport=tls>;auth_pass=xxxxxxxxxx

and got following error message:

> reg: sip:xxxxxxxxxx@sip.linphone.org: Register: Function not implemented

What could be the issue?
My terminal reports the same as xevilstar's except:

> Populated 3 contacts
> aufilt: gtk_vumeter
> Populated 2 audio codecs
> Populated 1 audio filter

**Reply by mbattista** [reply]
Example for account can be found here:
https://github.com/baresip/baresip/blob/main/docs/examples/accounts

**Reply by mbattista** [reply]
> clear instruction how to build the latest version of baresip-gtk

Clear instruction on how to build the latest baresip, which will also include the GTK module
```
sudo apt-get install build-essential libgtk-3-dev
git clone https://github.com/baresip/re.git
git clone https://github.com/baresip/baresip.git
cd re
cmake -B build
cmake --build build -j
sudo cmake --install build
sudo ldconfig
cd ..
cd baresip
cmake -B build
cmake --build build -j
sudo cmake --install build
sudo ldconfig
baresip
```

baresip should then startup.
if you do not see the phone in the top right, check in the ~/.baresip/config if you have the `module_app gtk.so` uncommented.

**Reply by ivanwmco** [reply]
Thank you so much, mbattista! I had already managed to get Sonetel and JMP Chat SIP accounts registered and working. I still would prefer to have the latest version installed. So, thanks again!

EDIT:
I went ahead, but ground to a halt at `cmake -B build`
Which was solved by `sudo apt install -y cmake`

Then the errors:
> Could NOT find OpenSSL

Fixed ([source](https://www.claudiokuenzler.com/blog/1216/could-not-find-openssl-error-compiling-cmake-source)):
`sudo apt install -y libssl-dev`
deleted the re and baresip GIT cloned folders and started over

Next:

> dl: mod: /usr/local/lib/baresip/modules/alsa.so (/usr/local/lib/baresip/modules/alsa.so: cannot open shared object file: No such file or directory)
> module alsa.so: No such file or directory [2]

Fixed ([source](https://github.com/baresip/baresip/issues/1093)): 
`sudo apt install -y libasound2-dev`
and recompiled 

Next:
> (baresip:33475): dconf-WARNING **: 16:02:21.769: unable to open file '/etc/dconf/db/local': /etc/dconf/db/local: invalid gvdb header; expect degraded performance

Fixed ([source](https://askubuntu.com/questions/1397205/how-to-solve-dconf-warning-unable-to-open-file-etc-dconf-db-site)):
`sudo dconf update`

And last but certainly not least:
> (baresip:33664): Gtk-CRITICAL **: 16:04:43.895: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed gtk_menu starting

The last error reflects that the GUI doesn't appear and I haven't found a solution yet that works.
- I tried installing 'AppIndicator and KStatusNotifierItem Support' in GNOME Extensions Manager.
- I tried `sudo apt install appmenu-gtk3-module`

**Reply by mbattista** [reply]
Ah I had a lot of those prerequisites already installed and forgot about them.

The error you are now getting is also known. Gnome and Wayland do not offer any systray.
https://github.com/baresip/baresip/issues/2295#issuecomment-1301079768

Starting baresip with "GDK_BACKEND=X11 baresip" should help.

**Reply by ivanwmco** [reply]
Thanks yet again, mbattista.

What worked for me (Ubuntu Desktop 22.04.4 LTS on NitroPad NV41) in the end:
```
sudo apt install -y build-essential libgtk-3-dev cmake libssl-dev libasound2-dev libcanberra-gtk-module libcanberra-gtk3-module
```
```
git clone https://github.com/baresip/re.git
```
```
git clone https://github.com/baresip/baresip.git
```
```
cd re
```
```
cmake -B build
```
```
cmake --build build -j
```
```
sudo cmake --install build
```
```
sudo ldconfig
```
```
cd
```
```
cd baresip
```
```
cmake -B build
```
```
cmake --build build -j
```
```
sudo cmake --install build
```
```
sudo ldconfig
```
```
sudo dconf update
```

Log out of the session. Log back in selecting 'Ubuntu on X.org' at the gear icon.

```
baresip
```
